__tests__/index.test.js

```diff
@@ -40,4 +41,82 @@ test('reject requests other than webhook payloads', async () => {
     context.req.headers['x-hub-signature-256'] = 'sha256=incorrect'
     context.req.rawBody = '# empty'
     await expectInvalidWebhook('Incorrect X-Hub-Signature')
+})
+
+let mockGetInstallationAccessToken = jest.fn(() => 'installation-access-token')
+jest.mock('../GitForWindowsHelper/get-installation-access-token', () => {
+    return mockGetInstallationAccessToken
+})
+
+let mockGitHubApiRequestAsApp = jest.fn()
+jest.mock('../GitForWindowsHelper/github-api-request-as-app', () => {
+    return mockGitHubApiRequestAsApp
+})
+
+let mockGitHubApiRequest = jest.fn((_context, _token, method, requestPath, payload) => {
+    if (method === 'POST' && requestPath.endsWith('/comments')) return {
+        id: -124,
+        html_url: `new-comment-url-${payload.body}`
+    }
+})
+jest.mock('../GitForWindowsHelper/github-api-request', () => {
+    return mockGitHubApiRequest
+})
+
+const makeContext = (body, headers) => {
+    const rawBody = JSON.stringify(body)
+    const sha256 = crypto.createHmac('sha256', process.env['GITHUB_WEBHOOK_SECRET']).update(rawBody).digest('hex')
+    return {
+        log: jest.fn(),
+        req: {
+            body,
+            headers: {
+                'content-type': 'application/json',
+                'x-hub-signature-256': `sha256=${sha256}`,
+                ...headers || {}
+            },
+            method: 'POST',
+            rawBody
+        }
+    }
+}
+
+const mockIssueComment = (comment) => {
+    return makeContext({
+        action: 'created',
+        comment: {
+            body: comment,
+            html_url: 'https://github.com/git-for-windows/git/issues/0',
+            id: 0,
+            user: {
+                login: 'statler and waldorf'
+            }
+        },
+        installation: {
+            id: 123
+        },
+        issue: {
+            number: 0
+        },
+        repository: {
+            name: 'git',
+            owner: {
+                login: 'git-for-windows'
+            }
+        }
+    }, {
+        'x-github-event': 'issue_comment'
+    })
+}
+
+test('test /hi', async () => {
+    const context = mockIssueComment('/hi')
+    expect(await index(context, context.req)).toBeUndefined()
+    expect(context.res).toEqual({
+        body: 'I said hi! new-comment-url-Hi @statler and waldorf!',
+        headers: undefined,
+        status: undefined
+    })
+    expect(mockGetInstallationAccessToken).toHaveBeenCalledTimes(1)
+    expect(mockGitHubApiRequestAsApp).not.toHaveBeenCalled()
```

Comment:
Should probably also `expect(mockGithubApiRequest).toHaveBeenCalledTimes(1)`

Reply:
Actually, there is that `.not.` part...

Reply:
That's for `mockGithubApiRequestAsApp`, I'm talking about an additional `expect()` for `mockGithubApiRequest`.

Reply:
Oh, but I think that should not be called because that call is "mocked away". Will check in a minute.

Reply:
Ah. It _is_ called. To add the issue comment.